### ISSUE TITLE
Error code: erroroidccall

### ISSUE THREAD

**Issue by EiffelD** [microsoft/o365-moodle#2358]
Hi @weilai-irl ,

We have a Moodle site where we had the secrete expire for the the oidc auth. We have a few sites and just update the secrete then off we go again. For this site we tried that but for some reason we get an error. 

This is similar to issue  [https://github.com/microsoft/o365-moodle/issues/1914](url). Below is the error with debugging on:

`Error in OpenID Connect. Please check logs for more information.

[More information about this error](https://docs.moodle.org/401/en/error/auth_oidc/erroroidccall)

Debug info:
Error code: erroroidccall×Dismiss this notification
Stack trace:
line 49 of /auth/oidc/classes/utils.php: moodle_exception thrown
line 345 of /auth/oidc/classes/oidcclient.php: call to auth_oidc\utils::process_json_response()
line 303 of /auth/oidc/classes/loginflow/authcode.php: call to auth_oidc\oidcclient->tokenrequest()
line 133 of /auth/oidc/classes/loginflow/authcode.php: call to auth_oidc\loginflow\authcode->handleauthresponse()
line 168 of /auth/oidc/auth.php: call to auth_oidc\loginflow\authcode->handleredirect()
line 31 of /auth/oidc/index.php: call to auth_plugin_oidc->handleredirect()`

I did the steps mentioned in the previus post and can see a 404 for /oidc

![image](https://github.com/microsoft/o365-moodle/assets/44169362/32d31112-900f-4174-883e-202b416fd26e)

Any help will be greatly appreciated!

Regards,
D

**Comment by weilai-irl** [comment]
Hi @EiffelD 

It looks like the response received when using the authorization code to exchange the access token is invalid. You will need to see what exactly is received - it's expected to be a json encoded string.

If adding debugging code on the site where this occurs is possible, you can simply var_dump the value of $response in the process_json_response() function. Otherwise, you can turn on "Record debug messages" setting in the "Other options" page of the auth_oidc plugin configuration, which will create an /auth_oidc/action_failed event in your standard log table with full details.

Once you have the details of the response, try to work out if it gives you any clue of the error, or simply post it back here if you want.

Regards,
Lai

**Comment by EiffelD** [comment]
Hi @weilai-irl ,

Just to add more info, the site is on 4.1.5+

below is a screenshot of when I hit the Provide Consent button:

![image](https://github.com/microsoft/o365-moodle/assets/44169362/1749b8e7-672e-46ff-a1af-1d6f4383cbfb)

**Comment by weilai-irl** [comment]
Hi @EiffelD 

Please follow my suggestions above to find out what exactly is returned by the access token request, which would hold the key to the issue.

Regards,
Lai

**Comment by EiffelD** [comment]
Hi @weilai-irl ,

Thanks for this information, will review once the data has been collected then come back to you if need be.

Regards,
D

**Comment by EiffelD** [comment]
Hi @weilai-irl ,

I had some help from another work colleague, we went to line 345 for the following file:/auth/oidc/classes/oidcclient.php. Added a print_r($this->endpoints);
        die;
above the line and shows the following:

Array ( [auth] => https://login.microsoftonline.com/domain.co.za/oauth2/authorize [token] => https://login.microsoftonline.com/domain.co.za/oauth2/token )

At one point it said the URL is blocked with the above address but now just shows the URL.  Any insight for the next steps will be appreciated. 

Regards,
D

**Comment by EiffelD** [comment]
Hi @weilai-irl ,

Hope you are doing well? I spent the better part of today trying to rebuild the app registration in Azure a few ways to see if it works. One thing I picked up is that if your Moodle domain is different to your Azure tenant domain, the PowerShell script won't and the correct Redirect URI and in a few other location for the App registration.

Any case that is a separate thing. No matter what I try I still end up at the same error:

_Error in OpenID Connect. Please check logs for more information.

[More information about this error](https://docs.moodle.org/401/en/error/auth_oidc/erroroidccall)

×Debug info:
Error code: erroroidccall
×Stack trace:
line 49 of /auth/oidc/classes/utils.php: moodle_exception thrown
line 347 of /auth/oidc/classes/oidcclient.php: call to auth_oidc\utils::process_json_response()
line 303 of /auth/oidc/classes/loginflow/authcode.php: call to auth_oidc\oidcclient->tokenrequest()
line 133 of /auth/oidc/classes/loginflow/authcode.php: call to auth_oidc\loginflow\authcode->handleauthresponse()
line 168 of /auth/oidc/auth.php: call to auth_oidc\loginflow\authcode->handleredirect()
line 31 of /auth/oidc/index.php: call to auth_plugin_oidc->handleredirect()_

I am not sure what to do next. Any advice?

Regards,
D

**Comment by EiffelD** [comment]
Hi @weilai-irl ,

Not exactly sure what happened but the SSO is working once again. The only thing we did was to add the microsoft.login address to the cURL block list in the System Admin > Security > HTTP. We then removed the address and also removed and added 443 to the cURL allowed ports in the same location.

Just to add, this problem happened when a secrete for the App registration for OIDC expired. We upgrated the Moodle server to 4.1 and it acted strange when trying to renew the secrete on the MS 365 integration.

Closing as resolved, thanks for your help.
